pkg/jwks/jwks.go

```diff
@@ -0,0 +1,249 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package jwks manages downloading and updating the keys from a JWKS source
+// for keys.
+package jwks
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/exposure-notifications-server/internal/database"
+	hadb "github.com/google/exposure-notifications-server/internal/verification/database"
+	"github.com/google/exposure-notifications-server/internal/verification/model"
+	"github.com/google/exposure-notifications-server/pkg/logging"
+	"github.com/rakutentech/jwk-go/jwk"
+)
+
+type Config struct {
+	Database database.Config
+}
+
+func (c *Config) DatabaseConfig() *database.Config {
+	return &c.Database
+}
+
+func (c *Config) JWKSManagerConfig() *Config {
+	return c
+}
+
+// Manager handles updating all HealthAuthorities if they've specified a JWKS
+// URI.
+type Manager struct {
+	ctxt context.Context
+	db   *database.DB
```

Comment:
I'd recommend storing the logger on the Manager instead of trying to get it from the context.

Reply:
done.